src/notifications/ldn.test.ts

```diff
@@ -378,28 +405,161 @@ describe("unstable_buildNotification", () => {
+  it("calls the included fetcher by default", async () => {
+    const mockedFetcher = jest.requireMock("../fetcher.ts") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo, RequestInit?]
+      >;
+    };
+    mockedFetcher.fetch.mockReturnValue(
+      Promise.resolve(
+        mockResponse("", {
+          headers: {
+            Link: '<../inbox>; rel="https://www.w3.org/ns/ldp#inbox"',
+            Location: "https://arbitrary.pod/resource",
+          },
+          url: "https://some.pod/",
+        })
+      )
+    );
+
+    await unstable_sendNotification(dataset(), "https://some.pod/resource");
+
+    expect(mockedFetcher.fetch).toHaveBeenCalled();
   });
 
-  it("should default to the fallback fetcher if no other is provided", async () => {
-    // TODO: Unimplemented
-    expect(null).toBeNull();
+  it("uses the given fetcher if provided", async () => {
+    const mockedResponse = mockResponse("", {
+      headers: {
+        Link: '<../inbox>; rel="https://www.w3.org/ns/ldp#inbox"',
+        Location: "https://arbitrary.pod/resource",
+      },
+      url: "https://some.pod/",
+    });
+    const mockFetch = jest
+      .fn(window.fetch)
+      .mockReturnValue(Promise.resolve(mockedResponse));
+
+    await unstable_sendNotification(dataset(), "https://some.pod/resource", {
+      fetch: mockFetch,
+    });
+
+    expect(mockFetch).toHaveBeenCalled();
   });
 
-  it("should send the provided notification to the inbox of the target resource", async () => {
+  it("should send the notification to the inbox of the given resource if found in a Link header", async () => {
+    const mockFetch = jest.fn(window.fetch).mockReturnValue(
+      Promise.resolve(
+        mockResponse("", {
+          status: 201,
+          headers: {
+            Link: '<../inbox>; rel="https://www.w3.org/ns/ldp#inbox"',
+            Location: "https://arbitrary.pod/inbox/notification_01",
+          },
+          url: "https://some.pod/",
+        })
+      )
+    );
     await unstable_sendNotification(
       dataset(),
-      Dataset.namedNode("https://your.pod/webId#you")
+      DataFactory.namedNode("https://some.pod/resource"),
+      {
+        fetch: mockFetch,
+      }
+    );
+    expect(mockFetch.mock.calls[1][0]).toEqual("https://some.pod/inbox");
+  });
+
+  it("should send the notification to the inbox of the given resource if found in its content", async () => {
+    const turtle = `
+      @prefix : <#>.
+      @prefix ldp: <https://www.w3.org/ns/ldp#>.
+
+      :aResource ldp:inbox :anInbox.
+    `;
+    const mockFetch = jest.fn(window.fetch).mockReturnValue(
+      Promise.resolve(
+        mockResponse(turtle, {
+          url: "https://some.pod/",
+          headers: {
+            Location: "https://some.pod/anInbox/notification",
+          },
+        })
+      )
+    );
+    await unstable_sendNotification(
+      dataset(),
+      DataFactory.namedNode("https://some.pod#aResource"),
```

Comment:
Not a bug, or a big deal or anything, but it seems strange to have resource IRIs with fragment identifiers - generally I'd reserve those for non-information resources, i.e. 'conceptual things'. Maybe this illustrates another advantage to using a shared vocab for test terms - i.e. to demonstrate best practices generally. Anyway, no need to change (unless it looks a little weird to you too).

Reply:
Actually another issue with this example IRI is that it doesn't clearly demonstrate containment - i.e. what's it's parent container...?! I assume it's the domain (i.e. `https://some.pod/`, which must end in a slash), so potential for confusion (or awkward questions!) maybe...

Reply:
I don't mind the long fragment, but you're right that showing containment might make things clearer. Fixed in b54c9f9adaee938686f543843a06fb4faf4a10db